src/components/mev-share/useSupportedBuilders.tsx

```diff
@@ -7,11 +7,33 @@ export type Builder = {
     "supported-apis": Array<string>, // TODO: can we please change this to camelCase
 }
 
-const fetchSupportedBuilders = async (): Promise<Array<Builder>> => {
-    const res = await axios.get("https://raw.githubusercontent.com/flashbots/dowg/main/builder-registrations.json")
-    return res.data
-}
 
-export const useSupportedBuilders = async (): Promise<Array<Builder>> => {
-    return useMemo(() => fetchSupportedBuilders(), [])
-}
+export const useSupportedBuilders = () => {
+    const [builders, setBuilders] = useState<Builder[]>([]);
+    const source = axios.CancelToken.source();
+
+    useEffect(() => {
```

Comment:
My understanding here is that you don't want the data request to be repeated in each rendering. IRRC `useMemo` is usually used to not repeat (memoize) expensive computation. Both works but feel that `useEffect` is more commonly used for side effect like data fetching

Reply:
yeah, the `useMemo` was intended to prevent the page from making an http call to get the builders every time the page loads. With `useEffect`, it will make the call every time the page re-renders, won't it?